### PR TITLE
src: explore: Add only-source only-header function

### LINKED ISSUE #184: Add rules for searching headers and c files
During the work with Kernel, it is common to search for functions and structs, as a result, we can find the occurrence of a string match in multiple files. However, there are situations that the developer only wants to take a look at the herder file or a C file. For this reason, it could be nice to have a specialized search for the header and C file. For example: 

```bash
kw e vkms_set_crc_source
drivers/gpu/drm/vkms/vkms_composer.c:236:int vkms_set_crc_source(struct drm_crtc *crtc, const char *src_name)
drivers/gpu/drm/vkms/vkms_crtc.c:161:   .set_crc_source         = vkms_set_crc_source,
drivers/gpu/drm/vkms/vkms_drv.h:145:int vkms_set_crc_source(struct drm_crtc *crtc, const char *src_name);

kw e -h vkms_set_crc_source
drivers/gpu/drm/vkms/vkms_drv.h:145:int vkms_set_crc_source(struct drm_crtc *crtc, const char *src_name);

kw e -c vkms_set_crc_source
drivers/gpu/drm/vkms/vkms_composer.c:236:int vkms_set_crc_source(struct drm_crtc *crtc, const char *src_name)
drivers/gpu/drm/vkms/vkms_crtc.c:161:   .set_crc_source         = vkms_set_crc_source,
```

### PR DESCRIPTION
**Pull request by hilookas**
This commit introduces two options `--only-source` and `--only-header`.

This commit also adds a new sample header file to test this function.

Relevant documentations are also updated.

Closes #184

Signed-off-by: Haiqin Cui <i@18kas.com>

### CODE REVIEW

**Comment by hilookas** [comment]
See also: #679

File: documentation/man/features/explore.rst [hunk 1]
```diff
@@ -37,3 +38,9 @@ OPTIONS
   ignores files inside .git, except if it is called inside .git directory. In
   other words, if you use this option you will notice that ``git grep`` is
   used first, and then GNU grep.
+
+-c | \--only-source:
+  With this option, it is possible to show only the results from the source.
+
+-h | \--only-header:
```
Comment (rodrigosiqueira) [inline documentation/man/features/explore.rst:45]: let's use -h for the help function to keep all parameters consistent.

File: src/explore.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@ function explore_main()
 {
   local ret
 
-  if [[ "$1" =~ -h|--help ]]; then
+  if [[ "$1" =~ --help ]]; then
```
Comment (rodrigosiqueira) [inline src/explore.sh:17]: -h is for a short help, and --help is for open the man page.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

File: src/explore.sh (hunk 2 above)
Comment (hilookas) [reply src/explore.sh:17]: What about "-H" for header "-h" for help? :D

**Comment by rodrigosiqueira** [comment]
Nice PR!
Let's use just the PR https://github.com/kworkflow/kworkflow/pull/679 , you just need to add the extra commit in that PR.

**Pull request closed without merging**